orbstation/code/antagonists/contract_killer/contract_killer_objectives.dm

```diff
@@ -0,0 +1,208 @@
+		/area/station/service/kitchen/diner,
+		/area/station/service/bar,
+		/area/station/service/library,
+		/area/station/service/chapel,
+		/area/station/service/theater,
+		/area/station/engineering/lobby,
+		/area/station/medical/break_room,
+		/area/station/security/detectives_office,
+		/area/station/cargo/boutique,
+		/area/station/cargo/sorting,
+		/area/station/cargo/lobby,
+		/area/station/science/lobby,
+		/area/station/science/robotics/lab,
+	))
+
+/datum/objective/contract_killer/location/New()
+	. = ..()
+	find_location()
+
+/datum/objective/contract_killer/location/proc/find_location()
+	var/list/possible_areas = GLOB.the_station_areas.Copy()
+	for(var/area/possible_area as anything in possible_areas)
+		if(!is_type_in_typecache(possible_area, whitelisted_areas))
+			possible_areas -= possible_area
+	target_location = pick(possible_areas)
+
+/// Leave the body somewhere
+/datum/objective/contract_killer/location/body
+	name = "leave body in location"
+	weight = CONTRACT_KILLER_OBJ_COMMON
+
+/datum/objective/contract_killer/location/body/update_explanation_text()
+	..()
+	if(target_location)
+		explanation_text = "Leave the body in the [initial(target_location.name)]."
+	else
+		explanation_text = "Leave the body where it will be found." //if we somehow fail to find a location, give a more generic version
+
+/datum/objective/contract_killer/location/head
+	name = "leave head in location"
+	weight = CONTRACT_KILLER_OBJ_RARE
+
+/datum/objective/contract_killer/location/head/update_explanation_text()
+	..()
+	if(target_location)
+		explanation_text = "Cut off the target's head and leave it in the [initial(target_location.name)]. Do what you want with the body."
+	else
+		explanation_text = "Cut off the target's head and leave it where it will be found. Do what you want with the body." //if we somehow fail to find a location, give a more generic version
+
+/// Objective to take a photo of the body and upload it to a newscaster.
+/datum/objective/contract_killer/newscaster
+	name = "newscaster taunt"
+	explanation_text = "Take a photo of the body and upload it anonymously to a newscaster. Summon and equip the provided ID card to keep it \
+		anonymous! Do as you like with the body when you finish."
+	weight = CONTRACT_KILLER_OBJ_COMMON
+
+/datum/objective/contract_killer/proc/give_summon_action()
+	if(owner?.current)
+		var/datum/action/summon_anonymous_id/summon_action = new(owner.current)
+		summon_action.Grant(owner.current)
+
+/datum/action/summon_anonymous_id
+	name = "Summon Anonymous ID"
+	desc = "Calls an anonymous ID into your hand. Equip this in your ID slot to make a newscast anonymously."
+	check_flags = AB_CHECK_CONSCIOUS | AB_CHECK_INCAPACITATED | AB_CHECK_HANDS_BLOCKED
+	button_icon = 'icons/obj/card.dmi'
+	button_icon_state = "card_grey"
+
+/datum/action/summon_anonymous_id/Trigger(trigger_flags)
+	. = ..()
+	var/obj/item/card/id/advanced/anonymous/new_card = new
+	owner.put_in_hands(new_card)
+	qdel(src)
```

Comment:
does this delete the action when you get it? just making sure that this is how this works

Reply:
Yeah, it does.

Reply:
neat!